casper/src/main/resources/PoS.rhox

```diff
@@ -366,104 +507,97 @@ in {
+                                    commitCurrentEpochRewards!(state, currentRewards, *updatedRewardsCh) |
+                                    for (@newCommittedState <- updatedRewardsCh) {
+                                      // 3. Update `withdrawers` according to `allBonds` and `pendingWithdrawers` 
+                                      // when pendingWithdrawers is not empty, remove the validator in `allBonds`
+                                      movePendingWithdrawer!(newCommittedState, *movePendingWithdrawCh) |
+                                      for (@newMovedPendingState <- movePendingWithdrawCh){
+                                        // 4. When the block number reaches the quantinue of `withdrawers` ,
+                                        //  transfer the corresponding bonds+ rewards to the corresponding validator from posVault. 
+                                        // 5. update `pendingWithdrawers` to empty Map {}.
+                                        removeQuarantinedWithdrawers!(blockNumber, newMovedPendingState, *removeQuarantinedCh) |
+                                        for (@newRemovedQuantinueState <- removeQuarantinedCh) {
+                                          // 6. Pick new `activeValidators`
+                                          pickActiveValidators!(newMovedPendingState.get("allBonds").toList(), *newValidatorsCh) |
+                                          for (@newValidators <- newValidatorsCh) {
+                                            stateUpdateCh!(newRemovedQuantinueState.set("activeValidators", newValidators), (true, Nil))
+                                          }
+                                        }
+                                      }
                                     }
                                   }
-                                } |
+                                } | 
                                 // Transfers withdrawing validator's bond + accumulated rewards from PoS vault to their vault.
-                                new payWithdrawer in {
-                                  contract payWithdrawer(@(pk, amount), returnCh) = {
-                                    new vaultCh, revAddressCh in {
-                                      revAddressOps!("fromPublicKey", pk, *revAddressCh) |
-                                      for (@toRevAddress <- revAddressCh) {
-                                        @posVault!("transfer", toRevAddress, amount, posAuthKey, *returnCh)
-                                      }
+                                contract payWithdrawer(@(pk, amount), returnCh) = {
+                                  new vaultCh, revAddressCh in {
+                                    revAddressOps!("fromPublicKey", pk, *revAddressCh) |
+                                    for (@toRevAddress <- revAddressCh) {
+                                      @posVault!("transfer", toRevAddress, amount, posAuthKey, *returnCh)
```

Comment:
It would be nice to create target vault so we don't ends up hanging if not created before.

Reply:
Make sense.